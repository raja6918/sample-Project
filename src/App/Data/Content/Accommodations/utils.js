import { DEFAULT_COST, DEFAULT_DURATION } from './Constants';

import {
  PER_CREW_MEMBER,
  USD,
  CHECKIN_CHECKOUT,
} from './../../../../_shared/configurationEntities';

import { getStringHours } from './../../../../_shared/helpers';

export const TRANSIT_TIME_MAX_VALUE = 2147483647;

const hasTrueValueObject = obj => {
  const Keys = Object.keys(obj);
  for (let i = 0; i < Keys.length; i++) {
    if (obj[Keys[i]]) {
      return true;
    }
  }
  return false;
};

const isThereATransitError = transitErrors => {
  const Keys = Object.keys(transitErrors);

  // Looping station codes
  for (let i = 0; i < Keys.length; i++) {
    const code = transitErrors[Keys[i]];
    const codeKeys = Object.keys(code);

    //Looping code properties
    for (let j = 0; j < codeKeys.length; j++) {
      if (typeof code[codeKeys[j]] === 'boolean' && code[codeKeys[j]]) {
        return true; // There is an error in transit section
      } else if (typeof code[codeKeys[j]] === 'object' && code[codeKeys[j]]) {
        //Loop through the internal array
        let isError = false;
        code[codeKeys[j]].forEach(item => {
          if (hasTrueValueObject(item)) {
            isError = true; //Errors in extra time array
          }
        });
        if (isError) {
          return true;
        }
      }
    }
  }
  return false; //No errors
};

const areValidTransitDetails = transitDetails => {
  const Keys = Object.keys(transitDetails);
  for (let i = 0; i < Keys.length; i++) {
    const code = transitDetails[Keys[i]];
    const { duration, extraTravelTimes, transportCost } = code;

    if (
      duration === '' ||
      duration === '-0' ||
      isNaN(duration) ||
      parseFloat(duration) < 0 ||
      duration > TRANSIT_TIME_MAX_VALUE
    ) {
      return false;
    }
    if (
      transportCost < 0 ||
      transportCost === '-0' ||
      isNaN(transportCost) ||
      transportCost === ''
    ) {
      return false;
    }
    for (let i = 0; i < extraTravelTimes.length; i++) {
      if (
        extraTravelTimes[i].extraTime === '' ||
        isNaN(extraTravelTimes[i].extraTime) ||
        parseFloat(extraTravelTimes[i].extraTime) < 0
      ) {
        return false;
      }
    }
  }
  return true;
};

export const isReadyToSubmit = (
  state,
  transitDetails,
  transitErrors,
  defaultTransists
) => {
  const {
    name,
    stations,
    typeCode,
    contractStartDate,
    contractLastDate,
    cost,
    currencyCode,
    billingPolicyCode,
    checkInTime,
    checkOutTime,
    errors,
  } = state;

  let isTransitSectionError = transitErrors
    ? isThereATransitError(transitErrors)
    : false;

  isTransitSectionError = defaultTransists ? false : isTransitSectionError;

  const isValidTransitDetails = defaultTransists
    ? true
    : areValidTransitDetails(transitDetails);

  const valid =
    name &&
    stations.length > 0 &&
    typeCode &&
    contractStartDate &&
    contractLastDate &&
    cost &&
    currencyCode &&
    isValidTransitDetails;

  const isCheckInOutTime = billingPolicyCode === CHECKIN_CHECKOUT;

  const areCheckInOutTimesValid = isCheckInOutTime
    ? checkInTime !== null && checkOutTime !== null
    : true;

  const noErrors =
    !errors.name &&
    !errors.cost &&
    !errors.extendedStayCostFactor &&
    !errors.capacity &&
    !isTransitSectionError;

  if (isCheckInOutTime) {
    return valid && noErrors && areCheckInOutTimesValid;
  } else {
    return valid && noErrors;
  }
};

export const countDecimals = value => {
  if (value.includes('.')) {
    return value.split('.')[1].length || 0;
  } else {
    return 0;
  }
};

export const evaluateRegex = (regex, term) => {
  const reg = new RegExp(regex);
  const isValidRegex = reg.test(term);
  return isValidRegex;
};

const getExtraTimesArray = extraTravelTimes => {
  let newExtraTimesArray = [];
  for (let i = 0; i < extraTravelTimes.length; i++) {
    newExtraTimesArray = [
      ...newExtraTimesArray,
      {
        duration: extraTravelTimes[i].extraTime,
        startTime: getStringHours(extraTravelTimes[i].startTime),
        endTime: getStringHours(extraTravelTimes[i].endTime),
      },
    ];
  }
  return newExtraTimesArray;
};

export const formatTransitDetails = transitDetails => {
  const Keys = Object.keys(transitDetails);
  let newTransitDetails = [];
  /* Looping through all the codes */
  for (let i = 0; i < Keys.length; i++) {
    newTransitDetails = [
      ...newTransitDetails,
      {
        billingPolicyCode: transitDetails[Keys[i]].billingPolicyCode,
        cost: +transitDetails[Keys[i]].transportCost,
        currencyCode: transitDetails[Keys[i]].transportCurrencyCode,
        duration: +transitDetails[Keys[i]].duration,
        extraTravelTimes: getExtraTimesArray(
          transitDetails[Keys[i]].extraTravelTimes
        ),
        stationCode: Keys[i],
      },
    ];
  }

  return newTransitDetails;
};

const formatTimeToDate = time => {
  const [getHours, getMinutes] = time.split(':');
  const date = new Date('10/10/2010').setHours(getHours);
  const time2 = new Date(date);

  time2.setMinutes(getMinutes);

  return time2;
};

const getInverseExtraTimeArray = extraTravelTimes => {
  let newExtraTimesArray = [];
  for (let i = 0; i < extraTravelTimes.length; i++) {
    newExtraTimesArray = [
      ...newExtraTimesArray,
      {
        extraTime: extraTravelTimes[i].duration,
        startTime: formatTimeToDate(extraTravelTimes[i].startTime),
        endTime: formatTimeToDate(extraTravelTimes[i].endTime),
      },
    ];
  }
  return newExtraTimesArray;
};

export const formatInverseTransitDetails = transitDetails => {
  let newTransitDetails = {};
  for (let i = 0; i < transitDetails.length; i++) {
    newTransitDetails = {
      ...newTransitDetails,
      [transitDetails[i].stationCode]: {
        billingPolicyCode: transitDetails[i].billingPolicyCode,
        transportCost: transitDetails[i].cost,
        transportCurrencyCode: transitDetails[i].currencyCode,
        duration: transitDetails[i].duration,
        extraTravelTimes: getInverseExtraTimeArray(
          transitDetails[i].extraTravelTimes
        ),
        stationCode: transitDetails[i].stationCode,
      },
    };
  }

  return newTransitDetails;
};

export const getDefaultTransitDetails = stations => {
  let newTransitDetails = [];
  for (let i = 0; i < stations.length; i++) {
    newTransitDetails = [
      ...newTransitDetails,
      {
        billingPolicyCode: PER_CREW_MEMBER,
        cost: DEFAULT_COST,
        currencyCode: USD,
        duration: DEFAULT_DURATION,
        extraTravelTimes: [],
        stationCode: stations[i].code,
      },
    ];
  }
  return newTransitDetails;
};

export const prepareCurrencyCodes = currencyCodes => {
  return currencyCodes.map(currencyCode => ({
    value: currencyCode,
    display: currencyCode,
  }));
};
