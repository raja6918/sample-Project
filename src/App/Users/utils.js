import pick from 'lodash/pick';
import { RegExs, ENTITY_FIELDS, getDefaultEntity } from './Constants';

export const validatePassword = data => {
  let containsUpperCase = false;
  let containsLowerCase = false;
  let contains8Chars = false;
  let containsNum = false;
  let containsNoBlank = false;

  // has uppercase letter
  if (data.toLowerCase() !== data) {
    containsUpperCase = true;
  }

  // has lowercase letter
  if (data.toUpperCase() !== data) {
    containsLowerCase = true;
  }

  // has number
  if (/\d/.test(data)) {
    containsNum = true;
  }

  // has 8 characters
  if (data.length >= 8) {
    contains8Chars = true;
  }

  // Contain no blank spaces
  if (/^\S{1,}$/.test(data)) {
    containsNoBlank = true;
  }

  return (
    containsLowerCase &&
    containsUpperCase &&
    contains8Chars &&
    containsNum &&
    containsNoBlank
  );
};

export const werePasswordFieldsFilled = (state, props) => {
  const { password, passwordRe } = state;
  const isPasswordActive = !state.disablePasswordField;
  const isCreate = !props.user;
  const areFilled =
    password !== '' && passwordRe !== '' && password === passwordRe;

  if (isCreate) return areFilled;

  return isPasswordActive ? areFilled : true;
};

export const wereAllFieldsFilled = (state, props) => {
  return ENTITY_FIELDS.reduce((acc, field) => {
    if (!acc) return acc;

    const value = state[field];

    if (field !== 'password') return acc && value !== '';
    return acc && werePasswordFieldsFilled(state, props);
  }, true);
};

export const hasError = errors => {
  return Object.values(errors).reduce((hasPrevErrors, hasError) => {
    return hasPrevErrors || hasError;
  }, false);
};

export const mapEntityToState = entity => ({
  ...getDefaultEntity(),
  ...pick(entity, ENTITY_FIELDS),
});

export const evaluateRegex = (regex, term) => {
  if (!regex) return true;

  const reg = new RegExp(regex);
  const isValidRegex = reg.test(term);
  return isValidRegex;
};

export const updateErrors = (key, value, state) => {
  const { errors } = state;

  if (value === '') {
    return {
      ...errors,
      [key]: false,
    };
  }

  if (key === 'password') {
    return {
      ...errors,
      [key]: !validatePassword(value),
    };
  }

  if (key === 'passwordRe') {
    return {
      ...errors,
      [key]: value !== state.password,
    };
  }

  return {
    ...errors,
    [key]: !evaluateRegex(RegExs[key], value),
  };
};

export const rolesFilter = (roles = []) => {
  return roles.filter(
    role => role.name !== 'uma_authorization' && role.name !== 'offline_access'
  );
};
