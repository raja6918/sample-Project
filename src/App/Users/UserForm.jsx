import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import pick from 'lodash/pick';
import get from 'lodash/get';
import PerfectScrollbar from 'react-perfect-scrollbar';
import 'react-perfect-scrollbar/dist/css/styles.css';

import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Collapse from '@material-ui/core/Collapse';
import MUIFormControl from '@material-ui/core/FormControl';

import ModelSelector from '../../components/ModelSelector';
import ErrorMessage from '../../components/FormValidation/ErrorMessage';
import Form from '../../components/FormDrawer/Form';
import FormHeader from '../../components/FormDrawer/FormHeader';
import FormBody from '../../components/FormDrawer/FormBody';
import PasswordValidator from '../../components/PasswordValidator/PasswordValidator';
import { perfectScrollConfig } from '../../utils/common';

import {
  mapEntityToState,
  updateErrors,
  wereAllFieldsFilled,
  hasError,
} from './utils';
import { ENTITY_FIELDS, getDefaultEntity } from './Constants';

const Input = styled(TextField)`
  .MuiFormLabel-root.Mui-disabled {
    color: rgba(0, 0, 0, 0.38) !important;
  }
  width: 100%;
  margin: 0;
  &.error > label {
    color: #d10000 !important;
  }
`;

const FormControl = styled(MUIFormControl)`
  width: 100%;
`;

const GridMarginTop = styled(Grid)`
  margin-top: 14px;
`;

const GridCheckbox = styled(Grid)`
  & span {
    font-size: 14px;
  }
  margin-bottom: -32px;
  margin-top: -14px;
`;

class UserForm extends Component {
  state = {
    ...getDefaultEntity(),
    passwordRe: '',
    disablePasswordField: false,
    selectedRole: false,
    isFormDirty: false,
    errors: {
      firstName: false,
      lastName: false,
      userName: false,
      email: false,
      password: false,
      passwordRe: false,
      roleId: false,
    },
  };

  componentWillReceiveProps(nextProps) {
    const hasToggleOpen = this.props.isOpen !== nextProps.isOpen;
    const hasUserChanged =
      get(this.props.user, 'id') !== get(nextProps.user, 'id');

    if (!hasToggleOpen && !hasUserChanged) return;

    const dataToUpdate = {
      ...mapEntityToState(nextProps.user),
      passwordRe: '',
      disablePasswordField: false,
      isFormDirty: false,
      errors: {
        firstName: false,
        lastName: false,
        userName: false,
        email: false,
        password: false,
        passwordRe: false,
        roleId: false,
      },
    };

    if (nextProps.user) {
      dataToUpdate.disablePasswordField = true;
    }

    this.setState(dataToUpdate);
  }

  tooglePasswordFields = () => {
    const { disablePasswordField, errors } = this.state;

    const dataToUpdate = {
      disablePasswordField: !disablePasswordField,
      password: '',
      passwordRe: '',
      errors: {
        ...errors,
        password: false,
        passwordRe: false,
      },
    };

    this.forcePasswordFieldsCleanUp();

    this.setState(dataToUpdate, this.checkFormIsReady);
  };

  forcePasswordFieldsCleanUp = () => {
    document.getElementById('password').value = '';
    document.getElementById('passwordRe').value = '';
    /* The below piece of code is in order to remove the active behavior of an input,
       MUI does this by default but unfortunately on this case does not
    */
    const Labels = document.querySelectorAll("label[for^='password']");
    for (let i = 0; i < Labels.length; i++) {
      const labelClass = Labels[i].getAttribute('class');
      const classes = labelClass.split(' ');
      for (let j = 0; j < classes.length; j++) {
        if (classes[j].includes('shrink')) {
          Labels[i].classList.remove(classes[j]);
        }
      }
    }
    /* End code's comments */
  };

  isReadyToSubmit() {
    return !this.state.isFormDirty;
  }

  handleInputChange = e => {
    this.onChange(e.target.name, e.target.value);
  };

  onChange = (fieldName, value, shouldValidForm = true) => {
    const errors = updateErrors(fieldName, value, this.state);
    const { roles } = this.props;
    const role = {};
    let valueToBeSelected = {
      errors,
      [fieldName]: value,
    };
    if (fieldName === 'roleId') {
      const selectedRole = roles.filter(role => role.id === value);
      role['role'] = selectedRole[0].roleName;
      valueToBeSelected = { ...valueToBeSelected, ...role };
    }
    this.setState(valueToBeSelected, () => {
      if (shouldValidForm) this.checkFormIsReady();
    });
  };

  checkFormIsReady = () => {
    const { isFormDirty } = this.state;
    const _hasError = hasError(this.state.errors);
    const allFieldsWereFilled = wereAllFieldsFilled(this.state, this.props);

    if (allFieldsWereFilled && !_hasError) {
      if (!isFormDirty) {
        this.setState({
          isFormDirty: true,
        });
      }
    } else {
      if (isFormDirty) {
        this.setState({
          isFormDirty: false,
        });
      }
    }
  };

  handleSubmit = formRef => {
    const entity = pick(this.state, ENTITY_FIELDS);
    if (this.props.user) entity.id = this.props.user.id;
    this.props.handleOk(formRef, entity);
  };

  render() {
    const { t, roles, user, ...rest } = this.props;
    const addUser = 'DIALOG.ADD_USER';
    const editUser = 'DIALOG.EDIT_USER';
    const errorMessages = 'ERRORS.USERS';
    const {
      errors,
      firstName,
      lastName,
      userName,
      email,
      password,
      passwordRe,
      roleId,
    } = this.state;

    return (
      <Form
        okButton={t(`${user ? editUser : addUser}.ok`)}
        isDisabled={this.isReadyToSubmit()}
        anchor={'right'}
        formId={'userform'}
        restrictReadOnly={true}
        {...rest}
        handleOk={this.handleSubmit}
      >
        <FormHeader>
          <span>{!user ? t(`${addUser}.title`) : t(`${editUser}.title`)}</span>
          <span>{`${firstName} ${lastName}`}</span>
        </FormHeader>
        <FormBody>
          <PerfectScrollbar option={perfectScrollConfig}>
            <div>
              <Grid container spacing={3}>
                <Grid item xs={12} sm={6}>
                  <FormControl error={errors.firstName}>
                    <Input
                      inputProps={{
                        name: 'firstName',
                        maxLength: 50,
                      }}
                      id="firstName"
                      label={t(`${addUser}.firstName`)}
                      margin="normal"
                      onChange={this.handleInputChange}
                      defaultValue={firstName}
                      className={errors.firstName ? 'error' : ''}
                      required
                    />
                    <ErrorMessage
                      isVisible={errors.firstName}
                      message={t(`${errorMessages}.firstName`)}
                    />
                  </FormControl>
                </Grid>
                <Grid item xs={12} sm={6}>
                  <FormControl error={errors.lastName}>
                    <Input
                      inputProps={{
                        name: 'lastName',
                        maxLength: 50,
                      }}
                      id="lastName"
                      label={t(`${addUser}.lastName`)}
                      margin="normal"
                      defaultValue={lastName}
                      onChange={this.handleInputChange}
                      className={errors.lastName ? 'error' : ''}
                      required
                    />
                    <ErrorMessage
                      isVisible={errors.lastName}
                      message={t(`${errorMessages}.lastName`)}
                    />
                  </FormControl>
                </Grid>
                <Grid item xs={12} sm={12}>
                  <FormControl error={errors.userName}>
                    <Input
                      inputProps={{
                        name: 'userName',
                        maxLength: 25,
                      }}
                      id="userName"
                      label={t(`${addUser}.userName`)}
                      margin="normal"
                      defaultValue={userName}
                      onChange={this.handleInputChange}
                      className={errors.userName ? 'error' : ''}
                      required
                      disabled={user}
                    />
                    <ErrorMessage
                      isVisible={errors.userName}
                      message={t(`${errorMessages}.userName`)}
                    />
                  </FormControl>
                </Grid>
                <Grid item xs={12} sm={12}>
                  <FormControl error={errors.email}>
                    <Input
                      inputProps={{
                        name: 'email',
                        maxLength: 100,
                      }}
                      id="email"
                      label={t(`${addUser}.email`)}
                      margin="normal"
                      defaultValue={email}
                      onChange={this.handleInputChange}
                      className={errors.email ? 'error' : ''}
                      required
                    />
                    <ErrorMessage
                      isVisible={errors.email}
                      message={t(`${errorMessages}.emailMalformed`)}
                    />
                  </FormControl>
                </Grid>
                {this.props.user && (
                  <GridCheckbox item xs={12} sm={12}>
                    <FormControlLabel
                      control={
                        <Checkbox
                          style={{ padding: 8 }}
                          color="secondary"
                          onChange={this.tooglePasswordFields}
                        />
                      }
                      name="changePassword"
                      label="Change password"
                      style={
                        !this.state.disablePasswordField
                          ? { paddingBottom: 22 }
                          : {}
                      }
                    />
                  </GridCheckbox>
                )}
                <Grid item xs={12} sm={12}>
                  <Collapse
                    in={!this.state.disablePasswordField || !this.props.user}
                  >
                    <Grid item xs={12} sm={12}>
                      <FormControl error={errors.password}>
                        <Input
                          inputProps={{
                            name: 'password',
                            type: 'password',
                            maxLength: 50,
                          }}
                          id="password"
                          label={
                            user
                              ? t(`${editUser}.password`)
                              : t(`${addUser}.password`)
                          }
                          margin="normal"
                          defaultValue={password}
                          onChange={this.handleInputChange}
                          required={
                            !this.state.disablePasswordField || !this.props.user
                          }
                        />
                        <PasswordValidator t={t} value={password} />
                      </FormControl>
                    </Grid>
                    <GridMarginTop item xs={12} sm={12}>
                      <FormControl error={errors.passwordRe}>
                        <Input
                          inputProps={{
                            name: 'passwordRe',
                            type: 'password',
                            maxLength: 50,
                          }}
                          id="passwordRe"
                          label={
                            user
                              ? t(`${editUser}.passwordRe`)
                              : t(`${addUser}.passwordRe`)
                          }
                          margin="normal"
                          defaultValue={passwordRe}
                          onChange={this.handleInputChange}
                          className={errors.passwordRe ? 'error' : ''}
                          required={
                            !this.state.disablePasswordField || !this.props.user
                          }
                        />
                        <ErrorMessage
                          isVisible={errors.passwordRe}
                          message={t(`${errorMessages}.passwordDoNotMatch`)}
                        />
                      </FormControl>
                    </GridMarginTop>
                  </Collapse>
                </Grid>
                <Grid item xs={12} sm={12}>
                  <FormControl>
                    <ModelSelector
                      label={t(`${addUser}.role`)}
                      name="roleId"
                      items={roles}
                      required={true}
                      selected={roleId}
                      menukey="id"
                      menuitemlabel="roleName"
                      setDefaultOption={false}
                      handleChange={value => this.onChange('roleId', value)}
                    />
                  </FormControl>
                </Grid>
              </Grid>
            </div>
          </PerfectScrollbar>
        </FormBody>
      </Form>
    );
  }
}

UserForm.propTypes = {
  user: PropTypes.shape({
    email: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    lastName: PropTypes.string.isRequired,
    password: PropTypes.string,
    role: PropTypes.string.isRequired,
    userName: PropTypes.string.isRequired,
  }),
  isOpen: PropTypes.bool.isRequired,
  handleCancel: PropTypes.func.isRequired,
  handleOk: PropTypes.func.isRequired,
  formId: PropTypes.string.isRequired,
  t: PropTypes.func.isRequired,
  roles: PropTypes.arrayOf(PropTypes.string).isRequired,
  onClose: PropTypes.func.isRequired,
};

UserForm.defaultProps = {
  user: null,
};

export default UserForm;
