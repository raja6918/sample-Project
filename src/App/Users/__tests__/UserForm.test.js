import React from 'react';
import { shallow } from 'enzyme';
import ModelSelector from '../../../components/ModelSelector';

import UserForm from '../UserForm';

import { evaluateRegex } from './../utils';

const users = [
  {
    id: 1001,
    firstName: 'Ruben',
    lastName: 'Flores',
    userName: 'rubnsom',
    email: 'ruben.flores@kronos.com',
    role: 'Administrator',
  },
  {
    id: 1002,
    firstName: 'Ramon',
    lastName: 'Barraza',
    userName: 'rbarraza',
    email: 'ramon.barraza@kronos.com',
    role: 'Administrator',
  },
];

const t = jest.fn();
const closeUserForm = jest.fn();
const handleUserForm = jest.fn();
const onExited = jest.fn();

const roles = ['Admin', 'Planner', 'Reviewer'];

test('UserForm Component renders', () => {
  const wrapper = shallow(
    <UserForm
      open={true}
      handleCancel={closeUserForm}
      handleOk={handleUserForm}
      t={t}
      onClose={closeUserForm}
      formId={'userForm'}
      roles={roles}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

test.skip('UserForm Component shows filled inputs with correct user', () => {
  const wrapper = shallow(
    <UserForm
      open={true}
      handleCancel={closeUserForm}
      handleOk={handleUserForm}
      t={t}
      onClose={closeUserForm}
      user={users[0]}
      formId={'userForm'}
      roles={roles}
    />
  );

  expect(wrapper.find('[id="firstName"]').prop('defaultValue')).toBe(
    users[0].firstName
  );
  expect(wrapper.find('[id="lastName"]').prop('defaultValue')).toBe(
    users[0].lastName
  );
  expect(wrapper.find('[id="userName"]').prop('defaultValue')).toBe(
    users[0].userName
  );
  expect(wrapper.find('[id="email"]').prop('defaultValue')).toBe(
    users[0].email
  );

  expect(wrapper.find(ModelSelector).prop('selected')).toBe(users[0].role);
});

test('UserForm Component shows Change Password when user prop is received', () => {
  const wrapper = shallow(
    <UserForm
      open={true}
      handleCancel={closeUserForm}
      handleOk={handleUserForm}
      t={t}
      onClose={closeUserForm}
      formId={'userForm'}
      roles={roles}
    />
  );

  expect(wrapper.state('disablePasswordField')).toBe(false);
  expect(wrapper.state('selectedRole')).toBe(false);

  wrapper.setProps({ user: users[0] });
  expect(wrapper.state('disablePasswordField')).toBe(true);

  wrapper.setProps({ user: null });
  expect(wrapper.state('disablePasswordField')).toBe(false);
});

test('state works correct', () => {
  const wrapper = shallow(
    <UserForm
      open={true}
      handleCancel={closeUserForm}
      handleOk={handleUserForm}
      t={t}
      onClose={closeUserForm}
      user={users[0]}
      formId={'userForm'}
      roles={roles}
      onExited={onExited}
    />
  );

  wrapper.instance().checkFormIsReady();
});

test('evaluate Regexp function works correct', () => {
  const wrapper = shallow(
    <UserForm
      open={true}
      handleCancel={closeUserForm}
      handleOk={handleUserForm}
      t={t}
      onClose={closeUserForm}
      user={users[0]}
      formId={'userForm'}
      roles={roles}
      onExited={onExited}
    />
  );

  let isValidName = evaluateRegex('^\\S(.*)\\S', 'Jose');
  expect(isValidName).toBe(true);
  isValidName = evaluateRegex('^\\S(.*)\\S', ' Jose');
  expect(isValidName).toBe(false);
});

test('handleBlur function works as expected', () => {
  const wrapper = shallow(
    <UserForm
      open={true}
      handleCancel={closeUserForm}
      handleOk={handleUserForm}
      t={t}
      onClose={closeUserForm}
      user={users[0]}
      formId={'userForm'}
      roles={roles}
      onExited={onExited}
    />
  );

  const event = {
    target: {
      name: 'email',
      value: ' joe@example',
    },
  };

  const regex = '^[a-zA-Z0-9].*';

  wrapper.instance().handleInputChange(event, regex);
  const isWrongEmail = wrapper.state('errors').email;

  expect(isWrongEmail).toBe(true);
});
