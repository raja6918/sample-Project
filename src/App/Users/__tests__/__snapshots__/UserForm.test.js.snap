// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm Component renders 1`] = `
<Form
  anchor="right"
  cancelButton="Cancel"
  enableReadOnly={false}
  formId="userForm"
  handleCancel={[MockFunction]}
  handleOk={[Function]}
  handleSubmit={[Function]}
  isDisabled={true}
  okButton="Ok"
  onChange={[Function]}
  onClose={[MockFunction]}
  open={true}
  restrictReadOnly={true}
>
  <FormHeader>
    <span />
    <span>
       
    </span>
  </FormHeader>
  <FormBody>
    <ScrollBar
      className=""
      component="div"
      containerRef={[Function]}
      option={
        Object {
          "swipeEasing": true,
          "wheelSpeed": 0.4,
        }
      }
    >
      <div>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={6}
            xs={12}
          >
            <UserForm__FormControl
              error={false}
            >
              <UserForm__Input
                className=""
                defaultValue=""
                id="firstName"
                inputProps={
                  Object {
                    "maxLength": 50,
                    "name": "firstName",
                  }
                }
                margin="normal"
                onChange={[Function]}
                required={true}
              />
              <ErrorMessage
                error={true}
                isVisible={false}
                message=""
              />
            </UserForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={6}
            xs={12}
          >
            <UserForm__FormControl
              error={false}
            >
              <UserForm__Input
                className=""
                defaultValue=""
                id="lastName"
                inputProps={
                  Object {
                    "maxLength": 50,
                    "name": "lastName",
                  }
                }
                margin="normal"
                onChange={[Function]}
                required={true}
              />
              <ErrorMessage
                error={true}
                isVisible={false}
                message=""
              />
            </UserForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <UserForm__FormControl
              error={false}
            >
              <UserForm__Input
                className=""
                defaultValue=""
                disabled={null}
                id="userName"
                inputProps={
                  Object {
                    "maxLength": 25,
                    "name": "userName",
                  }
                }
                margin="normal"
                onChange={[Function]}
                required={true}
              />
              <ErrorMessage
                error={true}
                isVisible={false}
                message=""
              />
            </UserForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <UserForm__FormControl
              error={false}
            >
              <UserForm__Input
                className=""
                defaultValue=""
                id="email"
                inputProps={
                  Object {
                    "maxLength": 100,
                    "name": "email",
                  }
                }
                margin="normal"
                onChange={[Function]}
                required={true}
              />
              <ErrorMessage
                error={true}
                isVisible={false}
                message=""
              />
            </UserForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <WithStyles(ForwardRef(Collapse))
              in={true}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                sm={12}
                xs={12}
              >
                <UserForm__FormControl
                  error={false}
                >
                  <UserForm__Input
                    defaultValue=""
                    id="password"
                    inputProps={
                      Object {
                        "maxLength": 50,
                        "name": "password",
                        "type": "password",
                      }
                    }
                    margin="normal"
                    onChange={[Function]}
                    required={true}
                  />
                  <PasswordValidator
                    t={
                      [MockFunction] {
                        "calls": Array [
                          Array [
                            "DIALOG.ADD_USER.ok",
                          ],
                          Array [
                            "DIALOG.ADD_USER.title",
                          ],
                          Array [
                            "DIALOG.ADD_USER.firstName",
                          ],
                          Array [
                            "ERRORS.USERS.firstName",
                          ],
                          Array [
                            "DIALOG.ADD_USER.lastName",
                          ],
                          Array [
                            "ERRORS.USERS.lastName",
                          ],
                          Array [
                            "DIALOG.ADD_USER.userName",
                          ],
                          Array [
                            "ERRORS.USERS.userName",
                          ],
                          Array [
                            "DIALOG.ADD_USER.email",
                          ],
                          Array [
                            "ERRORS.USERS.emailMalformed",
                          ],
                          Array [
                            "DIALOG.ADD_USER.password",
                          ],
                          Array [
                            "DIALOG.ADD_USER.passwordRe",
                          ],
                          Array [
                            "ERRORS.USERS.passwordDoNotMatch",
                          ],
                          Array [
                            "DIALOG.ADD_USER.role",
                          ],
                        ],
                        "results": Array [
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                          Object {
                            "type": "return",
                            "value": undefined,
                          },
                        ],
                      }
                    }
                    value=""
                  />
                </UserForm__FormControl>
              </WithStyles(ForwardRef(Grid))>
              <UserForm__GridMarginTop
                item={true}
                sm={12}
                xs={12}
              >
                <UserForm__FormControl
                  error={false}
                >
                  <UserForm__Input
                    className=""
                    defaultValue=""
                    id="passwordRe"
                    inputProps={
                      Object {
                        "maxLength": 50,
                        "name": "passwordRe",
                        "type": "password",
                      }
                    }
                    margin="normal"
                    onChange={[Function]}
                    required={true}
                  />
                  <ErrorMessage
                    error={true}
                    isVisible={false}
                    message=""
                  />
                </UserForm__FormControl>
              </UserForm__GridMarginTop>
            </WithStyles(ForwardRef(Collapse))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={12}
            xs={12}
          >
            <UserForm__FormControl>
              <ModelSelector
                defaultValue=""
                disabled={false}
                handleChange={[Function]}
                items={
                  Array [
                    "Admin",
                    "Planner",
                    "Reviewer",
                  ]
                }
                label=""
                menuitemlabel="roleName"
                menukey="id"
                name="roleId"
                placeholder=""
                required={true}
                selected=""
                setDefaultOption={false}
              />
            </UserForm__FormControl>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </div>
    </ScrollBar>
  </FormBody>
</Form>
`;
